src/blocks.zig

```diff
@@ -414,23 +414,45 @@ pub const Block = packed struct { // MARK: Block
 
 // MARK: Tick
 pub var tickFunctions: utils.NamedCallbacks(TickFunctions, TickFunction) = undefined;
-pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void;
+pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32, _param: ?*anyopaque) void;
 pub const TickFunctions = struct {
-	pub fn replaceWithCobble(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void {
-		std.log.debug("Replace with cobblestone at ({d},{d},{d})", .{x, y, z});
-		const cobblestone = parseBlock("cubyz:cobblestone");
+	pub const FunctionName = enum {
+		null,
+		replaceWith,
+	};
```

Comment:
I don't like this design. It's mixing the downsides of enums (they are not runtime extensible → no mods) with the downside of function pointers (→runtime dispatch).

I'd suggest to just use a VTable for all functions related to TickEvents with a separate loadData function for each event type. This is how most other things are implemented, e.g. check out tool modifiers